src/api/routes/tx.ts

```diff
@@ -248,6 +251,12 @@ export function createTxRouter(db: DataStore): RouterWithAsync {
       return res.redirect('/extended/v1/tx/0x' + tx_id);
     }
 
+    try {
+      hexToBuffer(tx_id);
+    } catch (error: any) {
+      throw new InvalidRequestError(error.message, InvalidRequestErrorType.invalid_tx_id);
+    }
+
```

Comment:
This looks like it should work well. Can you apply similar logic for the pagination parsing errors? E.g.
```ts
let limit: number;
let offset: number;
try {
  const limit = parseTxQueryLimit(req.query.limit ?? 96);
  const offset = parsePagingQueryInput(req.query.offset ?? 0);
} catch (error) {
  throw new InvalidRequestError(error.message, InvalidRequestErrorType.invalid_tx_id);
}
```

And can you scan through the rest of the endpoints fof hex input values to apply the same validation?

Reply:
Sure! Thanks a lot for the thorough feedback.

Reply:
Also, in the case of the `parseTxQueryLimit` and `parsePagingQueryInput` functions, since those are explicitly used for route handling, they can directly throw the `InvalidRequestError` error